### PR TITLE
Fix creature vision through camera/mirror

### PR DESCRIPTION
**Pull request by alanbrady**
Way back since v3603, creatures have been able to see the player through
cameras and mirrors.  This does not make sense.

This fixes the issue by adding another cache to the map_cache which
contains only camera/mirror vision.  Thus player vision takes into
account both "normal" vision and camera vision. Meanwhile monsters using
the vision cache for forced symmetry only use the "normal" vision cache.

Fixes issues #13491.

SUMMARY: Bugfixes "Fix creatures being able to see you through cameras and mirrors"

### CODE REVIEW

File: src/lightmap.cpp [hunk 1]
```diff
@@ -832,6 +836,8 @@ void map::build_seen_cache( const tripoint &origin, const int target_z )
         }
     }
 
+    std::fill_n( &camera_cache[0][0], SEEX * MAPSIZE * SEEY * MAPSIZE, 0.0f );
```
Comment (alanbrady) [inline src/lightmap.cpp]: This is the one bit I'm not sure about.  If I take this out the camera cache never gets cleared, I can't figure out how the seen_cache gets cleared so I can duplicate that.

Comment (alanbrady) [reply src/lightmap.cpp]: Figured it out and fixed it.

File: src/lightmap.cpp [hunk 2]
```diff
@@ -456,9 +456,11 @@ bool map::pl_sees( const tripoint &t, const int max_range ) const
     }
 
     const auto &map_cache = get_cache_ref( t.z );
-    return map_cache.seen_cache[t.x][t.y] > LIGHT_TRANSPARENCY_SOLID + 0.1 &&
-        ( map_cache.seen_cache[t.x][t.y] * map_cache.lm[t.x][t.y] >
-          g->u.get_vision_threshold( map_cache.lm[g->u.posx()][g->u.posy()] ) ||
+    const float vis = map_cache.seen_cache[t.x][t.y] > map_cache.camera_cache[t.x][t.y] ?
```
Comment (kevingranade) [inline src/lightmap.cpp]: Use std::max() instead of a ternary.

**Comment by kevingranade** [comment]
For context, the caveat to this is here:
https://github.com/CleverRaven/Cataclysm-DDA/blob/62b54d82fd47b5ba8ac8cbc418f8ab5fed4d75bd/src/creature.cpp#L189-L195
When creatures (including NPCs) check visibility to the player, they use seen_cache, but not camera_cache.

**Comment by Lorith** [comment]
Awesome, this bug has been a major pain for me for ages.  Glad someone figured it out :)

**Comment by paulenka-aleh** [comment]
So that if intelligent (intelligent enough) foe is in the vehicle near the mirror / camera display and player is behind the vihicle, player will be invisible?

**Comment by Zireael07** [comment]
I don't think there's any 'intelligent enough' foes in game - animals don't recognize things in mirrors, and Zs are just dumb 'straight to target' enemies.

**Comment by paulenka-aleh** [comment]
How is it going to work with NPCs? Though seems to be quite a rare case...

**Pull request merged**